src/lancie-admin-endpoint/lancie-admin-pages/lancie-admin-types/lancie-admin-ticket-types.html

```diff
@@ -0,0 +1,245 @@
+        <paper-button on-tap="tryChangeBuyableOfTypes">Confirm</paper-button>
+      </div>
+    </lancie-dialog>
+
+  </template>
+  <script>
+    (function () {
+      'use strict';
+
+      Polymer({
+        is: 'lancie-admin-ticket-types',
+        properties: {
+          options: Object,
+          data: Object,
+          mappings: {
+            type: Object,
+            value: function() {
+              return {
+              };
+            }
+          },
+          renderers: {
+            type: Object,
+            value: function() {
+              return {
+                possibleOptions: function(data) {
+                  var text = data.reduce(function(text, t){
+                    return text + t.name + ' ' + t.price +', ';
+                  }, '');
+                  return text.substr(0, text.length - 2);
+                },
+              };
+            }
+          }
+        },
+
+        onAddTypeResponse: function(e, request) {
+          if (request.succeeded) {
+            this.$.addTicketTypeDialog.close();
+            this.$.getTypeAjax.generateRequest();
+          } else {
+            this.$.addDialogError.setError("Could not add ticket type. Please refresh and try again.");
+          }
+        },
+
+        onTypeResponse: function(e, request) {
+          if (request.succeeded) {
+            this.$.table.setData(request.response);
+          } else {
+            this.$.optionError.setError('Could not retrieve ticket types. Please refresh and try again.');
+          }
+        },
+
+        openAddTypeDialog: function() {
+          this.$.addTicketTypeForm.reset();
+          this.$.addTicketTypeDialog.open();
+        },
+
+        tryAddType: function(e, request) {
+          this.$.addTicketTypeForm.validateAndSubmit();
+          this.$.addTicketTypeForm.reset();
+        },
+
+        openAddOptionToTypeDialog: function() {
+          if (this.$.table.selected().length > 0) {
+            this.selectedTypes = this.$.table.selected();
+            this.$.noTypeSelectedError.clear();
+            this.$.typeSelected.render();
+            this.$.addOptionToTypeDialog.open();
+          } else {
+            this.$.noTypeSelectedError.setError('Please select a type first.')
+          }
+        },
+
+        optionSelected: function(e) {
+          this.selectedOption = this.options.filter(option => {
+            return option.name === e.target.selectedItem.innerText;
+          })[0];
+        },
+
+        tryAddOptionToTypes: function() {
+          this.selectedTypes.forEach(t => this.tryAddOptionToType(t))
+          this.$.table.clearSelection();
+          this.$.addOptionToTypeDialog.close();
+          this.$.getTypeAjax.generateRequest();
+        },
+
+        tryAddOptionToType: function(t) {
+          this.selectedId = t.id;
+          this.$.ajaxAddOptionToType.generateRequest();
+        },
+
+        openChangeBuyableDialog: function() {
```

Comment:
Reverse the condition here. Also, do the indexes start at `1`?

Reply:
Indexes start at 0, but when only index 0 is filled length is still 1

Reply:
Oops sorry, that's embarrassing